.github/workflows/pytest.yaml

```diff
@@ -0,0 +1,36 @@
+---
+name: pytest
+
+on:
+  pull_request:
+    branches:
+      - main
+jobs:
+  pytest:
+    runs-on: ubuntu-latest
+
+    timeout-minutes: 15
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install .
+      - name: Test with pytest
+        run: |
+          pytest test \
+            --doctest-modules \
+            --junitxml=junit/test-results.xml
+      - name: Upload pytest test results
+        uses: actions/upload-artifact@v3
+        with:
```

Comment:
Consider using `if-no-files-found: error` if pytest fails in this way for some reason.

Reply:
Good call - that's a new feature for me. Done with 62a70d8.